### ISSUE TITLE
Antigen and sourcing from different scripts

### ISSUE THREAD

**Issue by ming13** [zsh-users/antigen#600]
I have [such setup](https://github.com/ming13/dotfiles/tree/891974a5c3fdda17c8db99a55f2a14680b88aac0/zsh). 

Basically on my Mac Zsh initialization does the following.

* Iterate through Darwin scripts, one of which sources `antigen.zsh`.
* Iterate through Unix scripts, first of which executes `antigen` commands.

Unfortunately it stopped to work and I don’t know which Antigen version worked well, sorry.

Steps.

1. `rm -rf ~/.antigen ~/.zcompdump`
1. Open new Terminal tab.
1. None of Antigen bundles are loaded, i. e. no extra highlighting or completions. `antigen list` shows bundles though.
1. `mv ~/.zsh/unix/_plugins.zsh ~/.zsh/darwin/plugins_after.zsh`
1. `rm -rf .antigen .zcompdump`
1. Open new Terminal tab.
1. Everything works as expected.

In other words, if I change script location containing `antigen` calls everything starts to work fine.

**Comment by desyncr** [comment]
It seems the issue comes from the fact that it runs on a function scope at two different times. I'll continue to investigate.

Thanks for reporting.

**Comment by ming13** [comment]
@desyncr, sorry to bother you, but are there any news on this?
